docs/configuration/configuration.md

```diff
@@ -2257,6 +2261,78 @@ paths:
 
 Serverset data must be in the JSON format, the Thrift format is not currently supported.
 
+### `<stackit_sd_config>`
+
+[STACKIT](https://www.stackit.de/de/) SD configurations allow retrieving
+scrape targets from various APIs.
+
+One of the following `<stackit_role>` types can be configured to discover targets:
+
+#### `server`
+
+The `server` role discovers scrape targets per [IAAS API](https://docs.api.eu01.stackit.cloud/documentation/iaas/version/v1#tag/Servers/operation/v1ListServersInProject).
```

Comment:
maybe ad something about the address preference/order, like in https://prometheus.io/docs/prometheus/latest/configuration/configuration/#ovhcloud_sd_config e.g.

Reply:
It's possible that a service have zero IP addresses up 4.

I have to cover the case where a server does not have an IP address up there are 4 IP adresse only only one of for have a public IP assigned. 

The currently logic may looks a bit complicated, but cover some cases as well.